src/Tracing/GuzzleTracingMiddleware.php

```diff
@@ -29,39 +30,44 @@ public static function trace(?HubInterface $hub = null): Closure
 
                 $spanContext = new SpanContext();
                 $spanContext->setOp('http.client');
-                $spanContext->setData([
-                    'url' => (string) $request->getUri(),
-                    'method' => $request->getMethod(),
-                    'request_body_size' => $request->getBody()->getSize(),
-                ]);
                 $spanContext->setDescription($request->getMethod() . ' ' . $request->getUri());
 
                 $childSpan = $span->startChild($spanContext);
 
                 $request->withHeader('sentry-trace', $childSpan->toTraceparent());
 
-                $handlerPromiseCallback = static function ($responseOrException) use ($childSpan) {
+                $handlerPromiseCallback = static function ($responseOrException) use ($hub, $request, $childSpan) {
+                    $childSpan->finish();
```

Comment:
It cannot since it's a span, not a transaction so we are all good there :)

Only transactions themself finish by sending, span's do not they just set a timestamp so doing it as early as possible seems the most "fair" in regards to timing.

Reply:
>It cannot since it's a span, not a transaction so we are all good there :)

Right, I thought that `getSpan()` could also return a transaction because a transaction inherits the `Span` class, but it doesn't seem the case according to the code. It looks a bit strange to me that `finish()` is not the last operation of the middleware, but I don't see any downside in leaving it as-is so it's your call